src/main/java/com/okta/sdk/framework/ApiClient.java

```diff
@@ -45,18 +48,23 @@
     private AtomicReference<HttpResponse> lastHttpResponseRef = new AtomicReference<HttpResponse>();
 
     ////////////////////////////////////////////
-    // CONSTRUCTORS
+    // CONSTRUCTOR
     ////////////////////////////////////////////
 
     public ApiClient(ApiClientConfiguration config) {
-        this(HttpClientBuilder.create()
-                        .setUserAgent("OktaSDKJava_v" + Utils.getSdkVersion())
-                        .disableCookieManagement().build(),
-                config);
-    }
-
-    protected ApiClient(HttpClient httpClient, ApiClientConfiguration config) {
-        this.httpClient = httpClient;
+        PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+        connectionManager.closeExpiredConnections();
+        connectionManager.closeIdleConnections(5, TimeUnit.SECONDS); // Our TTL is 5 seconds
+
+        // The connections automatically expire/stale in 5000ms by default so we do not need to set the connection time out.
+        // When the connection expires/goes stale and sometimes when the httpClient is not aware, the next subsequent call fails which make the client
+        // realize that the connection has expired. The request can then be retried by the httpClient.
+        //https://hc.apache.org/httpcomponents-client-ga/tutorial/html/connmgmt.html Section 2.5
```

Comment:
Please do not link to the documentation. We can talk more about this if you have questions.